src/layers/basic.jl

```diff
@@ -461,38 +462,45 @@ This layer is often used to store word embeddings and retrieve them using indice
 
 ## Arguments
 
-  - `in_dims`: number of input dimensions
+  - `in_dims`: number(s) of input dimensions
   - `out_dims`: number of output dimensions
 
 ## Keyword Arguments
 
   - `init_weight`: initializer for the weight matrix
-    (`weight = init_weight(rng, out_dims, in_dims)`)
+    (`weight = init_weight(rng, out_dims, in_dims...)`)
 
 ## Input
 
   - Integer OR
   - Abstract Vector of Integers OR
-  - Abstract Array of Integers
+  - Abstract Array of Integers OR
+  - Tuple of Integers OR
+  - Tuple of Abstract Vectors of Integers OR
+  - Tuple of Abstract Arrays of Integers
 
 ## Returns
 
   - Returns the embedding corresponding to each index in the input. For an N dimensional
     input, an N + 1 dimensional output is returned.
   - Empty `NamedTuple()`
 """
-@concrete struct Embedding <: AbstractExplicitLayer
-    in_dims::Int
+@concrete struct Embedding{N} <: AbstractExplicitLayer
+    in_dims::NTuple{N, Int}
```

Comment:
This is probably not needed, just relax the `::Int`

Reply:
Done, thanks.

Reply:
I guess it would be also possible to restrict the type of `in_dims` a little bit:

```julia
@concrete struct Embedding{T <: Union{Int, NTuple{<:Any, Int}}} <: AbstractExplicitLayer
    in_dims::T
    out_dims::Int
    init_weight
end
```

But I completely relaxed the type as you said, and let `@concrete` take care of it:

```julia
@concrete struct Embedding <: AbstractExplicitLayer
    in_dims
    out_dims::Int
    init_weight
end
```